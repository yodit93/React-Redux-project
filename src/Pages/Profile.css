/* eslint-disable */
/* stylelint-disable */
* {
  padding: 0;
  margin: 0;
  box-sizing: border-box;
}

.profile-container {
  padding: 0 4%;
  display: grid;
  gap: 5%;
  grid-template-columns: 1fr 1fr;
}

.mission-lists,
.my-rockets {
  margin-bottom: 0.5rem;
}

.rocket-list, .mission-list {
  list-style: none;
  display: flex;
  flex-direction: column;
}

.rocket-item, .mission-item {
  display: flex;
  width: 100%;
  border: 1px solid #ccc;
  height: 5rem;
  border-radius: 0.2rem;
  align-items: center;
  padding: 0.5rem;
}

.rocket-name, .mission-name {
  font-size: 1.5rem;
  font-weight: 600;
}
