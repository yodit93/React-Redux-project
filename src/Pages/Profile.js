import { useSelector } from 'react-redux';
import React from 'react';
import './Profile.css';

function Profile() {
  const { rockets } = useSelector((store) => store.rockets);
  const { missions } = useSelector((store) => store.missions);
  return (
    <div className="profile-container">
      <div className="profile-missions-section">
        <h3 className="mission-lists">My Missions</h3>
        <ul className="mission-list">
          {missions.filter((mission) => mission.joined).map((mission) => (
            <li key={mission.id} className="mission-item">
              <span className="mission-name">{mission.name}</span>
            </li>
          ))}
        </ul>
      </div>
      <div className="profile-rockets-section">
        <h3 className="my-rockets">My Rockets</h3>
        <ul className="rocket-list">
          {rockets.filter((rocket) => rocket.reserved).map((rocket) => (
            <li key={rocket.id} className="rocket-item">
              <span className="rocket-name">{rocket.name}</span>
            </li>
          ))}
        </ul>
      </div>
    </div>
  );
}

export default Profile;
