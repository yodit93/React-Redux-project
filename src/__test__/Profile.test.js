import { render, screen } from '@testing-library/react';
import '@testing-library/jest-dom';
import renderer from 'react-test-renderer';
import { Provider } from 'react-redux';
import configureMockStore from 'redux-mock-store';
import Profile from '../Pages/Profile';

const mockStore = configureMockStore();
const store = mockStore({
  rockets: {
    rockets: [
      {
        id: '1',
        name: 'Falcon 1',
        description: 'The Falcon 1 was an expendable launch system privately developed and manufactured by SpaceX during 2006-2009',
        image: 'https://images2.imgbox.com/3c/0e/T8iJcSN3_o.png',
        reserved: false,
      },
    ],
    isLoading: false,
    error: null,
  },
  missions: {
    missions: [
      {
        id: '1',
        name: 'Falcon 1',
        description: 'The Falcon 1 was an expendable launch system privately developed and manufactured by SpaceX during 2006-2009',
      },
    ],
    isLoading: false,
    error: null,
  },
});

describe('Profile', () => {
  it('should render the Profile component', () => {
    const tree = renderer.create(<Provider store={store}><Profile /></Provider>).toJSON();
    expect(tree).toMatchSnapshot();
  });
  it('should have a text "My Missions"', () => {
    render(
      <Provider store={store}>
        {' '}
        <Profile />
      </Provider>,
    );
    const headingElement = screen.getByText(/My Missions/i);
    expect(headingElement).toBeInTheDocument();
  });
});
