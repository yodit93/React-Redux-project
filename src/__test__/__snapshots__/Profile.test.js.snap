// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Profile should render the Profile component 1`] = `
<div
  className="profile-container"
>
  <div
    className="profile-missions-section"
  >
    <h3
      className="mission-lists"
    >
      My Missions
    </h3>
    <ul
      className="mission-list"
    />
  </div>
  <div
    className="profile-rockets-section"
  >
    <h3
      className="my-rockets"
    >
      My Rockets
    </h3>
    <ul
      className="rocket-list"
    />
  </div>
</div>
`;
